src/dcap_provider.cpp

```diff
@@ -861,24 +1085,82 @@ static quote3_error_t convert_to_intel_error(sgx_plat_error_t platformError)
 
 static std::string build_pck_crl_url(
     std::string crl_name,
-    std::string api_version)
+    std::string api_version,
+    sgx_prod_type_t prod_type = SGX_PROD_TYPE_SGX)
 {
-    std::string version = get_collateral_version();
+    std::string version;
+    std::stringstream url;
+    std::string client_id;
+    if (prod_type == SGX_PROD_TYPE_TDX)
+    {
+        version = get_collateral_version_tdx();
+
+        client_id = get_client_id();

```

Comment:
move outside else if as both are using the same. For sgx client id can be appended after API version

Reply:
Fixed